stock_card_report/wizard/stock_card_report_wizard.py

```diff
@@ -0,0 +1,143 @@
+    )
+    location_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Location",
+        required=True,
+    )
+    initial_stock = fields.Float(
+        string='Initial Stock',
+    )
+    line_ids = fields.One2many(
+        comodel_name='stock.card.report.wizard.line',
+        inverse_name='explosion_id',
+    )
+
+    @api.model
+    def _prepare_line(self, stock_card, type, balance, location):
+        if type == 'in':
+            return {
+                'explosion_id': self.id,
+                'product_id': stock_card.product_id.id,
+                'location_id': location,
+                'date': stock_card.date,
+                'reference': stock_card.reference,
+                'product_in': stock_card.product_uom_qty,
+                'product_out': 0.00,
+                'product_balance': balance,
+            }
+        elif type == 'out':
+            return {
+                'explosion_id': self.id,
+                'product_id': stock_card.product_id.id,
+                'location_id': location,
+                'date': stock_card.date,
+                'reference': stock_card.reference,
+                'product_in': 0.00,
+                'product_out': stock_card.product_uom_qty,
+                'product_balance': balance,
+            }
+
+    @api.multi
+    def find_stock_card(self):
+        self.ensure_one()
+        line_obj = self.env['stock.card.report.wizard.line']
+
+        def _find_initial(initial_stock_card, location):
+            initial = 0.00
+            for line in initial_stock_card:
+                if line.location_dest_id.id == location:
+                    initial = initial + line.product_uom_qty
+                elif line.location_id.id == location:
+                    initial = initial - line.product_uom_qty
+            return initial
+
+        initial_stock_card = self.env['stock.move'].search([
+            '|', ('location_id', 'child_of', self.location_id.id),
+            ('location_dest_id', 'child_of', self.location_id.id),
+            ('product_id', '=', self.product_id.id),
+            ('date', '<', self.date_from)])
+        self.initial_stock = _find_initial(initial_stock_card, self.location_id.id)
+
+        def _create_lines(stock_card, location, initial_stock):
+            balance = initial_stock
+            for line in stock_card:
+                if line.location_dest_id.id == location:
+                    balance = balance + line.product_uom_qty
+                    vals = self._prepare_line(line, 'in', balance, location)
+                    line_obj.create(vals)
+                elif line.location_id.id == location:
+                    balance = balance - line.product_uom_qty
+                    vals = self._prepare_line(line, 'out', balance, location)
+                    line_obj.create(vals)
+
+        stock_card = self.env['stock.move'].search([
+            '|', ('location_id', 'child_of', self.location_id.id),
```

Comment:
https://www.odoo.com/forum/help-1/question/domain-notation-using-multiple-and-nested-and-2170